core/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -60,16 +61,11 @@ public static PrioritizedEsThreadPoolExecutor newSinglePrioritizing(String name,
 
     public static EsThreadPoolExecutor newScaling(String name, int min, int max, long keepAliveTime, TimeUnit unit, ThreadFactory threadFactory, ThreadContext contextHolder) {
         ExecutorScalingQueue<Runnable> queue = new ExecutorScalingQueue<>();
-        // we force the execution, since we might run into concurrency issues in offer for ScalingBlockingQueue
```

Comment:
How I wish I wrote down what we found out at the time. Now that  I look at the ExecutorScalingQueue, I see:

```
@Override
        public boolean offer(E e) {
            if (!tryTransfer(e)) {
                int left = executor.getMaximumPoolSize() - executor.getCorePoolSize(); <-- **** 
                if (left > 0) {
                    return false;
                } else {
                    return super.offer(e);
                }
            } else {
                return true;
            }
        }
```

The place I marked with `***` feels weird as it effectively returns false all the time, which is fine due to the fact that we want to prefer adding threads first and then, if it fails queue. Which is why we need the rejection policy. My guess is that this there for the case where people make a "fixed" thread pool out of a scaling one but having min==max (feels like an unneeded optimization to me, btw).

Do you agree, and if so, can we add a comment documenting?